spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -431,6 +462,18 @@ private static void importUnpartitionedSparkTable(SparkSession spark, TableIdent
       List<DataFile> files = TableMigrationUtil.listPartition(
           partition, Util.uriToString(sourceTable.location()), format.get(), spec, conf, metricsConfig, nameMapping);
 
+      if (checkDuplicateFiles) {
+        Dataset<Row> importedFiles = spark.createDataset(
+            Lists.transform(files, f -> f.path().toString()), Encoders.STRING()).toDF("file_path");
```

Comment:
I think this covers most cases, but probably will still have an issue for the odd Authority issues. Like if I have the table where my metadata entries have one authority or no authority, and my new files list doesn't. This is probably still as close as we can get without actually resolving everything I think ....